lib/discourse_js_processor.rb

```diff
@@ -100,19 +103,21 @@ def self.skip_module?(data)
   class Transpiler
     @mutex = Mutex.new
     @ctx_init = Mutex.new
+    @processor_mutex = Mutex.new
 
     def self.mutex
       @mutex
     end
 
-    def self.load_file_in_context(ctx, path, wrap_in_module: nil)
-      contents = File.read("#{Rails.root}/app/assets/javascripts/#{path}")
-      contents = <<~JS if wrap_in_module
-          define(#{wrap_in_module.to_json}, ["exports", "require", "module"], function(exports, require, module){
-            #{contents}
-          });
-        JS
-      ctx.eval(contents, filename: path)
+    def self.generate_js_processor
+      @processor_mutex.synchronize do
+        if Rails.env.development? || Rails.env.test?
+          !File.exist?("#{Rails.root}/app/assets/javascripts/compiled-js-processor.js")
+          error =
+            `yarn --silent esbuild --log-level=warning --bundle app/assets/javascripts/js-processor.js --external:fs --define:process='{"env":{}}' --outfile=#{JS_PROCESSOR_PATH}`.strip
```

Comment:
I think we should use `Discourse::Utils.execute_command` here. It has a number of safety-nets including timeouts, command injection protections, and will check exit codes.